packages/wonder-blocks-typography/src/typography.js

```diff
@@ -106,121 +111,171 @@ export class HeadingXSmall extends Component {
-    props: Props;
+    props: HeadingProps;
+
+    static defaultProps = {
+        tag: "span",
+    };
 
     render() {
+        const {tag, style, children} = this.props;
         return (
-            <Text style={[styles.BodySerifBlock, this.props.style]}>
-                {this.props.children}
+            <Text tag={tag} style={[styles.BodySerifBlock, style]}>
+                {children}
             </Text>
         );
     }
 }
 
 export class BodySerif extends Component {
-    props: Props;
+    props: HeadingProps;
+
+    static defaultProps = {
+        tag: "span",
+    };
 
     render() {
+        const {tag, style, children} = this.props;
         return (
-            <Text style={[styles.BodySerif, this.props.style]}>
-                {this.props.children}
+            <Text tag={tag} style={[styles.BodySerif, style]}>
+                {children}
             </Text>
         );
     }
 }
 
 export class BodyMonospace extends Component {
-    props: Props;
+    props: HeadingProps;
+
+    static defaultProps = {
+        tag: "span",
+    };
 
     render() {
+        const {tag, style, children} = this.props;
         return (
-            <Text style={[styles.BodyMonospace, this.props.style]}>
-                {this.props.children}
+            <Text tag={tag} style={[styles.BodyMonospace, style]}>
+                {children}
             </Text>
         );
     }
 }
 
 export class Body extends Component {
-    props: Props;
+    props: HeadingProps;
+
+    static defaultProps = {
+        tag: "span",
+    };
 
     render() {
+        const {tag, style, children} = this.props;
         return (
-            <Text style={[styles.Body, this.props.style]}>
-                {this.props.children}
+            <Text tag={tag} style={[styles.Body, style]}>
+                {children}
             </Text>
         );
     }
 }
 
 // TODO(kevinb): consider making labels block level elements
 export class LabelLarge extends Component {
-    props: Props;
+    props: HeadingProps;
+
+    static defaultProps = {
+        tag: "span",
+    };
 
     render() {
+        const {tag, style, children} = this.props;
         return (
-            <Text style={[styles.LabelLarge, this.props.style]}>
-                {this.props.children}
+            <Text tag={tag} style={[styles.LabelLarge, style]}>
+                {children}
             </Text>
         );
     }
 }
 
 export class LabelMedium extends Component {
-    props: Props;
+    props: HeadingProps;
+
+    static defaultProps = {
+        tag: "span",
+    };
 
     render() {
+        const {tag, style, children} = this.props;
         return (
-            <Text style={[styles.LabelMedium, this.props.style]}>
-                {this.props.children}
+            <Text tag={tag} style={[styles.LabelMedium, style]}>
+                {children}
             </Text>
         );
     }
 }
 
 export class LabelSmall extends Component {
-    props: Props;
+    props: HeadingProps;
```

Comment:
I would probably choose a name that is a bit more semantic.  `HeadingProps` seems too specific to `H` components.  Even if the types aliases. What do you think?

Reply:
Yep.  In fact, there is no reason to separate `HeadingProps` from `Props` anymore.  It should just be `Props` like it was before Alex's change.